libs/api/dev/src/lib/api-dev.ts

```diff
@@ -0,0 +1,36 @@
+import { formatApiErrorResponse, validateApiKey } from '@nmit-coursition/api/utils'
+import { type ApiErrorCode, ERROR_LIST } from '@nmit-coursition/api/utils'
+import { Elysia, t } from 'elysia'
+
+const errorResponseType = t.Object({
+  state: t.String(),
+  message: t.String(),
+  code: t.Number(),
+  errorCode: t.String(),
+  correlationId: t.String(),
+  description: t.Optional(t.String()),
+})
+
+export const apiDev = new Elysia({ prefix: '/dev' })
+  .guard({
+    headers: t.Object({
+      authorization: t.String({ error: 'You must provide API key to use this service.' }),
+    }),
+    response: {
+      401: errorResponseType,
```

Comment:
**thought:** I was thinking if we should repeat type as you did or use ElysiaJS suggested `model` approach.
https://elysiajs.com/essential/structure.html#model

What do you think? I have my opinion, but don't want to pollute yours, so I'll share it after you. :)